models/dataset.go

```diff
@@ -173,6 +172,18 @@ func (d *Dataset) RemoveContributor(role string, i int) error {
 	return nil
 }
 
+func (d *Dataset) GetUserContributorRoles(user *Person) string {
```

Comment:
this shouldn't be a method on dataset, it belongs in a view helper or in the io18n dictionary